# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.h
# Commit: 7b3b51a5efea
# Full Hash: 7b3b51a5efea5e04bdef378f4fa3940b77dc62f8
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-04-16 03:47:33
# Description:
#   Bug 1890077 - Ensure we check for a null target when creating paths for Canvas2D. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D207464
# ==============================================================================

diff -r 342488f68e70 -r 7b3b51a5efea dom/canvas/CanvasRenderingContext2D.h
--- a/dom/canvas/CanvasRenderingContext2D.h	Mon Apr 15 20:04:39 2024 +0000
+++ b/dom/canvas/CanvasRenderingContext2D.h	Mon Apr 15 20:11:09 2024 +0000
@@ -364,14 +364,18 @@
   }
 
   void ClosePath() {
-    EnsureWritablePath();
+    if (!EnsureWritablePath()) {
+      return;
+    }
 
     mPathBuilder->Close();
     mPathPruned = false;
   }
 
   void MoveTo(double aX, double aY) {
-    EnsureWritablePath();
+    if (!EnsureWritablePath()) {
+      return;
+    }
 
     mozilla::gfx::Point pos(ToFloat(aX), ToFloat(aY));
     if (!pos.IsFinite()) {
@@ -383,13 +387,17 @@
   }
 
   void LineTo(double aX, double aY) {
-    EnsureWritablePath();
+    if (!EnsureWritablePath()) {
+      return;
+    }
 
     LineTo(mozilla::gfx::Point(ToFloat(aX), ToFloat(aY)));
   }
 
   void QuadraticCurveTo(double aCpx, double aCpy, double aX, double aY) {
-    EnsureWritablePath();
+    if (!EnsureWritablePath()) {
+      return;
+    }
 
     mozilla::gfx::Point cp1(ToFloat(aCpx), ToFloat(aCpy));
     mozilla::gfx::Point cp2(ToFloat(aX), ToFloat(aY));
@@ -408,7 +416,9 @@
 
   void BezierCurveTo(double aCp1x, double aCp1y, double aCp2x, double aCp2y,
                      double aX, double aY) {
-    EnsureWritablePath();
+    if (!EnsureWritablePath()) {
+      return;
+    }
 
     BezierTo(mozilla::gfx::Point(ToFloat(aCp1x), ToFloat(aCp1y)),
              mozilla::gfx::Point(ToFloat(aCp2x), ToFloat(aCp2y)),
@@ -680,7 +690,7 @@
 
   /* This function ensures there is a writable pathbuilder available
    */
-  void EnsureWritablePath();
+  bool EnsureWritablePath();
 
   // Ensures a path in UserSpace is available.
   void EnsureUserSpacePath(
